lisp/mastodon-toot.el

```diff
@@ -92,34 +110,31 @@ Remove MARKER if RM is non-nil."
 
 Set `mastodon-toot--reply-to-id' to nil.
 Set `mastodon-toot--content-warning' to nil."
-  (kill-buffer-and-window)
-  (setq mastodon-toot--reply-to-id     nil
-        mastodon-toot--content-warning nil))
+  (kill-all-local-variables)
+  (kill-buffer-and-window))
 
 (defun mastodon-toot--cancel ()
-  "Kill new-toot buffer/window. Does not POST content to Mastodon."
+  "Kill new-toot buffer/window.  Does not POST content to Mastodon."
   (interactive)
   (mastodon-toot--kill))
 
-(defun mastodon-toot--remove-docs ()
-  "Get the body of a toot from the current compose buffer."
-  (let ((re "^|=+=|$"))
-    (save-excursion
-      (goto-char 0)
-      (re-search-forward re nil nil 2)
-      (buffer-substring (+ 2 (point)) (+ 1 (length (buffer-string)))))))
+(defun mastodon-toot--get-from-buffer ()
+  "Use the data stored in MASTODON-TOOT--EDIT-DATA and the compose buffer to extract the toot text."
+  (let ((toot-start (alist-get 'start mastodon-toot--edit-data))
```

Comment:
Since you are not using the `default` functionality of `alist-get` you can swap it out with something like `(cdr (assq key alist))`